veps/sig-network/hotpluggable-nad-ref.md

```diff
@@ -0,0 +1,227 @@
+This in turn calls various sub-logics that check for changes in areas like disk, volume, network, memory etc and decide if that requires a restart. The sub-logic for network [[2]](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/vmliveupdate/restart.go#L32)
+compares the interface and network specs of the VMI (currently running instance) to the VM (changed by the user), and asks the controller to add `RestartRequired` if any field (including the NAD reference field) in the network specs have changed [[3]](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/vmliveupdate/restart.go#L62).
+
+#### Copying specs from VM to VMI
+The same Sync cycle then copies various specs like network [[4]](https://github.com/kubevirt/kubevirt/blob/e661fdf24887592d74275c1c5ae9e5e1a1438433/pkg/virt-controller/watch/vm/vm.go#L3195), firmware etc from the VM to the VMI.
+The logic that deals with copying the interfaces/network specs [[5]](https://github.com/kubevirt/kubevirt/blob/7c016a3706abe6fd6627ae141ff7e04f4f0c7d81/pkg/network/controllers/vm.go#L110) checks if interfaces are being hotplugged or modified
+and copies the interfaces from VM to VMI specs accordingly [[6]](https://github.com/kubevirt/kubevirt/blob/7c016a3706abe6fd6627ae141ff7e04f4f0c7d81/pkg/network/controllers/vm.go#L147)
+It keeps the network specs unchanged.
+
+#### Adding `Migration` flags
+During the [updateStatus cycle](https://github.com/kubevirt/kubevirt/blob/77818370e6c1b932ec2a693256b399fd289dda1b/pkg/virt-controller/watch/vmi/lifecycle.go#L232), the VMI controller in virt-controller evaluates the migration state [[7]](https://github.com/kubevirt/kubevirt/blob/77818370e6c1b932ec2a693256b399fd289dda1b/pkg/virt-controller/watch/vmi/lifecycle.go#L1063).
+Changes to interfaces and networks can trigger 3 states - immediate/pending/no migration [[8]](https://github.com/kubevirt/kubevirt/blob/65460586b6a0910eb2b70098de627240a3790183/pkg/network/migration/evaluator.go#L85).
+In case of hotplug/unplug of interfaces with supported bindings it returns either immediate migration flag (for SIOV interfaces) or pending migration (which is a delayed migration - providing enough time for the `Dynamic Network Controller` to do an `InPlace` hotplug if applicable).
+This currently does not send back immediate/pending migration status in case of change in the NAD reference.
+
+
+### Proposed architecture
+
+The VEP proposes to trigger migration on changing the NAD reference (`networkName`), rather than requiring restart.
+
+#### Do not add RestartRequired flag for changes to networkName field
+The sub logic for network [[3]](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/vmliveupdate/restart.go#L62) that asks the VM Controller to add RestartRequired flag specs would be updated to exclude
+changes to networkName.
```

Comment:
The VM controller's logic will be adjusted to not add the `RestartRequired` condition in case the `networkName` field has changed.